OpenSim/Common/Component.cpp

```diff
@@ -192,6 +192,48 @@ void Component::finalizeFromProperties()
 
     markPropertiesAsSubcomponents();
     componentsFinalizeFromProperties();
+
```

Comment:
Do we need still need

```
  if (getName().empty()) {
         setName(IO::Lowercase(getConcreteClassName()) + "_");		         
     }		     
```

above? That seems redundant now.

Could you also document why this is the right place to put this code in `finalizeFromProperties()`? At first glance, it would seem preferable to put this new code above `markPropertiesAsSubcomponents()` (but I understand that you need to do that first to access the subcomponents).

Reply:
It is redundant in that we could have `_0`, `_1`, `_2`, ... names for unnamed components if we do not first at the class name. I prefer having the class name appear in the no name case (e.g. pathwrap_#). I can remove the now unnecessary trailing underscore, though.